syntax = "proto3";
package consumer;

// Service
service kafkaConsumer {
  rpc consumer(ConsumerRequest) returns (stream ConsumerResponse) {}
  rpc getRunningConsumers(getRunningConsumersRequest) returns (getRunningConsumersResponse) {}
  rpc stopAllRunningConsumers(stopAllRunningConsumersRequest) returns (stopAllRunningConsumersResponse) {}
  rpc stopRunningCosumer(stopRunningCosumerRequest) returns (stopRunningCosumerResponse) {}
}

// Request Name
message ConsumerRequest {
  string topicName=1;
  string group=2;
  int64 limit=3;
}

message getRunningConsumersRequest{
  string data=1;
}

message stopAllRunningConsumersRequest{
  string data=1;
}

message stopRunningCosumerRequest{
  string data=1;
}


// Response Array
message ConsumerResponse{
  bytes data=1;
}

message getRunningConsumersResponse{
  bytes data=1;
}

message stopAllRunningConsumersResponse{
  string data=1;
}

message stopRunningCosumerResponse{
  string data=1;
}
